packages/code-du-travail-frontend/src/piwik.js

```diff
@@ -1,21 +1,67 @@
-import ReactPiwik from "react-piwik";
 import { Router } from "../routes";
 import getConfig from "next/config";
 
 const {
   publicRuntimeConfig: { PIWIK_URL, PIWIK_SITE_ID }
 } = getConfig();
-
 if (typeof window !== "undefined" && PIWIK_URL && PIWIK_SITE_ID) {
-  const piwik = new ReactPiwik({
+  initPiwik({
     url: PIWIK_URL,
-    siteId: PIWIK_SITE_ID,
-    trackErrors: true
+    siteId: PIWIK_SITE_ID
   });
-  ReactPiwik.push(["trackPageView"]);
+  matopush(["trackPageView"]);
+  let previousPath;
   Router.events.on("routeChangeComplete", path => {
-    piwik.track({ path });
+    if (previousPath === path) {
+      return;
+    }
+    setTimeout(() => {
```

Comment:
We alrady are inside a ".on" listener, do we really need to wait for the next event loop to trigger this one ?

Reply:
this PR is still a draft but yeah I will add a comment if we keep the `setTimeout`